### ISSUE TITLE
[Bug] BackgroundColor of ContentPage not being updated on AppThemeBinding update

### ISSUE THREAD

**Issue by MyLastWish** [dotnet/maui#5681]
### Description

BackgroundColor of ContentPage does not update on app theme change, while most other colors do.

### Steps to Reproduce

1. Checkout this repo: https://github.com/MyLastWish/-MauiBackgroundColorBug
2. Run project.
3. When clicking the button it will update text color, but not page's backgroundColor.

### Version with bug

Preview 14 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows, I was *not* able test on other platforms

### Affected platform versions

Android API 30, Windows 10 Educational

### Did you find any workaround?

No.

**Comment by MyLastWish** [comment]
> Verified with Visual Studio Enterprise 2022 17.2.32317.152 on Windows and Android. When clicking Button, both text color and page's backgroundColor are not updated.
> 
> @MyLastWish Can you check your code again?

I have checked, it is not working as you said. I could have sworn, that it worked before uploading to repo, but that's another issue then - TextColor on Label has issue with it too.

Anyway, I have updated repository, this time it's text's background that correctly changes color, and I have checked it on windows and android. Both Label's and Page's background color are set using `BackgroundColor` property
